docs/source/library-user-guide/adding-udfs.md

```diff
@@ -55,62 +55,138 @@ of arguments.
 This a lower level API with more functionality but is more complex, also documented in [`advanced_udf.rs`].
 
 ```rust
+use std::sync::Arc;
 use std::any::Any;
+use std::sync::LazyLock;
 use arrow::datatypes::DataType;
+use datafusion_common::cast::as_int64_array;
 use datafusion_common::{DataFusionError, plan_err, Result};
-use datafusion_expr::{col, ColumnarValue, Signature, Volatility};
+use datafusion_expr::{col, ColumnarValue, Documentation, ScalarFunctionArgs, Signature, Volatility};
+use datafusion_expr::scalar_doc_sections::DOC_SECTION_MATH;
+use datafusion::arrow::array::{ArrayRef, Int64Array};
 use datafusion_expr::{ScalarUDFImpl, ScalarUDF};
 
+/// This struct for a simple UDF that adds one to an int32
 #[derive(Debug)]
 struct AddOne {
-    signature: Signature
-};
+  signature: Signature,
+}
 
 impl AddOne {
-    fn new() -> Self {
-        Self {
-            signature: Signature::uniform(1, vec![DataType::Int32], Volatility::Immutable)
-        }
-    }
+  fn new() -> Self {
+    Self {
+      signature: Signature::uniform(1, vec![DataType::Int32], Volatility::Immutable),
+     }
+  }
 }
 
+static DOCUMENTATION: LazyLock<Documentation> = LazyLock::new(|| {
```

Comment:
Actually @comphead  and @Omega359  made some nicer ways to create this documentation with macros (we can fix this as a follow on PR)

Reply:
I see, I'd like to explore the macro, could you share the name of that macro?

Reply:
It is called `user_doc`

Here is an example: https://github.com/apache/datafusion/blob/bd7daffe0c8643f008689f097eb887f09d063e7e/datafusion/functions/src/string/btrim.rs#L37-L59

Reply:
Okay updated and replaced it with `user_doc`, 
in order to make it work, I added 
```
datafusion-macros = { workspace = true }
datafusion-doc = { workspace = true }
```
to `datafusion/core/Cargo.toml`, not sure if this is the correct way though